src/vm/functions/iterables.rs

```diff
@@ -134,18 +135,21 @@ pub fn special_append(args: &[SymbolicExpression], env: &mut Environment, contex
             let element =  eval(&args[1], env, context)?;
             let ListData { mut data, type_signature } = list;
             let (entry_type, size) = type_signature.destruct();
-            runtime_cost!(cost_functions::APPEND, env, entry_type.size())?;
+            let element_type = TypeSignature::type_of(&element); 
+            runtime_cost!(cost_functions::APPEND, env,
+                          u64::from(cmp::max(entry_type.size(), element_type.size())))?;
             if entry_type.is_no_type() {
                 assert_eq!(size, 0);
-                Value::list_from(vec![ element ])
-            } else if !entry_type.admits(&element) {
-                Err(CheckErrors::TypeValueError(entry_type, element).into())
-            } else {
-                let next_type_signature = ListTypeData::new_list(entry_type, size + 1)?;
+                return Value::list_from(vec![ element ])
+            }
+            if let Ok(next_entry_type) = TypeSignature::least_supertype(&entry_type, &element_type) {
+                let next_type_signature = ListTypeData::new_list(next_entry_type, size + 1)?;
```

Comment:
Should we use some safe math on `size + 1`?

Reply:
We shouldn't need to --- the maximum clarity value size (which size cannot be greater than) is less than u32::max.